test/wait-task/pkg/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/controller"
+)
+
+var validRef = &v1alpha1.TaskRef{
+	APIVersion: "example.dev/v0",
```

Comment:
can we use something more descriptive here than `example.dev`? 

maybe `wait.tekton.dev/v1alpha1`?

Reply:
this will also be useful if/when we decide to add other custom tasks for testing

Reply:
I'm under the impression taht the new controller is meant to be the only one that is used for testing? say if we have a new feature to be tested, simply adds a new fields and related func in the reconciler? I'm okay with having more descriptive name here, probably customtask.tekton.test?

Reply:
agreed that it's the only controller and we'd add functionality to it...but that could change in the future, for example if we want to test a custom task that has a CRD (instead of this one that doesn't have a CRD)

even if we remain with this as the only custom task, it's still helpful to make the name more descriptive than "example"

Reply:
SG! To reflect what we discussed offline: We propose to recommend people to use `*.testing.tekton.dev/<version>` when defining new resources for testing purpose. For this particular case, we'll use `wait.testing.tekton.dev/v1alpha1`.

Reply:
And I'm not sure the right place to document the naming convention, do you think it's a good idea to bring it up in the pipeline working group? or probably doc WG?

Reply:
you can bring it up in the pipelines wg next week, but it shouldn't be a blocker for this PR because this not user-facing